### PR TITLE
[SPARK-38181][SS][DOCS] Update comments in KafkaDataConsumer.scala

### PR DESCRIPTION
**Pull request by ArvinZheng**
### What changes were proposed in this pull request?
Fixed some minor format issue in the code comments and rephrase some of them to make it more clear

### Why are the changes needed?
Minor format correction and better readability

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Not needed, no real code changes

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @HeartSaVioR

**Comment by ArvinZheng** [comment]
@dongjoon-hyun @HeartSaVioR updated the branch and resolved the conflict, please help take a look when you get a chance, thanks in advance.

**Comment by itholic** [comment]
Seems like the test still failing. Could you rebase?

@HeartSaVioR FYI

**Comment by ArvinZheng** [comment]
thanks @itholic for checking, just updated my branch, and yes, it's all doc/comment fixes

File: connector/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/consumer/KafkaDataConsumer.scala [hunk 1]
```diff
@@ -298,9 +296,10 @@ private[kafka010] class KafkaDataConsumer(
       s"requested $offset")
 
     // The following loop is basically for `failOnDataLoss = false`. When `failOnDataLoss` is
-    // `false`, first, we will try to fetch the record at `offset`. If no such record exists, then
-    // we will move to the next available offset within `[offset, untilOffset)` and retry.
-    // If `failOnDataLoss` is `true`, the loop body will be executed only once.
+    // `false`, we will try to fetch the record at `offset`, if the record does not exist, we will
+    // try to fetch next available record within [offset, untilOffset).
+    // If `failOnDataLoss` is `true`, the loop body will be executed only once, either return the
+    // record at `offset` or throw an exception when the record does not exist
```
Comment (anishshri-db) [inline connector/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/consumer/KafkaDataConsumer.scala:302]: Nit: Period at the end

**Review by anishshri-db — APPROVED** [review]
LGTM

**Comment by anishshri-db** [comment]
Seems like the tests are failing though ? Maybe try merging back and re-trigger ?

**Comment by HyukjinKwon** [comment]
@ArvinZheng mind rebasing this PR? seems like something went wrong about finding GA actions in your fork.

**Comment by HyukjinKwon** [comment]
Oh, also please enable GItHub Actions in your forked repository (https://github.com/ArvinZheng/spark). Apache Spark repository leverages PR author's GitHub Actions resources.

File: connector/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/consumer/KafkaDataConsumer.scala (hunk 1 above)
Comment (ArvinZheng) [reply connector/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/consumer/KafkaDataConsumer.scala]: thanks, updated.

**Review by HeartSaVioR — APPROVED** [review]
+1, sorry for the late reviewing.

**Pull request closed without merging**